com.unity.visualeffectgraph/Editor/GraphView/Elements/Controllers/VFXDataAnchorController.cs

```diff
@@ -123,6 +125,18 @@ protected override void ModelChanged(UnityEngine.Object obj)
             UpdateInfos();
             Profiler.EndSample();
 
+            if (m_GizmoContext != null)
+            {
+                HashSet<VFXExpression> expressions = new HashSet<VFXExpression>();
+                model.GetExpressions(expressions);
+
+                var currentExpressionHashCode = UIUtilities.GetHashCode(expressions);
+                if (currentExpressionHashCode != m_expressionHashCode)
+                {
+                    RefreshGizmo();
```

Comment:
I would add a comment here (for later purpose, if we are able to get more information from ModelChanged) :
```suggestion
                    //The ModelChanged will be triggered when a parent value is modified, detecting if there is a change in graph (link or change space) using expression hash to prevent the invalidation all cached gizmos
                    RefreshGizmo();
```
Feel free to edit or ignore.

Reply:
Done, with slightly different message